source/modulo_components/include/modulo_components/ComponentInterface.hpp

```diff
@@ -235,11 +251,20 @@ T ComponentInterface<NodeT>::get_parameter_value(const std::string& name) const
 }
 
 template<class NodeT>
-void
-ComponentInterface<NodeT>::add_parameter(const std::shared_ptr<state_representation::ParameterInterface>& parameter) {
-  parameter_map_.set_parameter(parameter);
+void ComponentInterface<NodeT>::add_parameter(
+    const std::shared_ptr<state_representation::ParameterInterface>& parameter, const std::string& description,
+    bool read_only
+) {
   auto ros_param = modulo_new_core::translators::write_parameter(parameter);
-  NodeT::declare_parameter(parameter->get_name(), ros_param.get_parameter_value());
+  if (!NodeT::has_parameter(parameter->get_name())) {
+    parameter_map_.set_parameter(parameter);
+    rcl_interfaces::msg::ParameterDescriptor descriptor;
+    descriptor.description = description;
+    descriptor.read_only = read_only;
```

Comment:
If we set this read_only property, what happens if you're trying to change the value?

Reply:
Ah yes, I meant to add a test for this but forgot. Essentially the value cannot be changed after it is initialised (so it should be stuck with whatever is in the NodeOptions, or otherwise the default value). I marked the period as read-only, and it might be interesting to have that option for other parameters too. The behaviour is described [here](https://github.com/ros2/rcl_interfaces/blob/4be87b33fb6aef6ea5ec5a0662c53f923f9fe9be/rcl_interfaces/msg/ParameterDescriptor.msg#L21-L22).